Updating rust toolchain to >2021-11-23 not possible

With toolchain `nightly-2021-11-23` the following error occurs:

```
error[E0133]: call to unsafe function is unsafe and requires unsafe function or block
   --> /.cargo/git/checkouts/incubator-teaclave-sgx-sdk-f208006cdf70a903/1e8a03a/sgx_tstd/src/sys_common/wtf8.rs:828:9
    |
828 |         next_code_point(&mut self.bytes).map(|c| CodePoint { value: c })
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ call to unsafe function
```

--- a/sgx_tstd/src/sys_common/wtf8.rs
+++ b/sgx_tstd/src/sys_common/wtf8.rs
@@ -825,7 +825,9 @@ impl<'a> Iterator for Wtf8CodePoints<'a> {
 
     #[inline]
     fn next(&mut self) -> Option<CodePoint> {
-        next_code_point(&mut self.bytes).map(|c| CodePoint { value: c })
+        unsafe {
+            next_code_point(&mut self.bytes).map(|c| CodePoint { value: c })
+        }
     }
 
     #[inline]

